File: Atcoder_#318/javascript/b_overlapping_sheets.js
function main(input) {
  const lines = input.split('\n');
  const N = parseInt(lines[0], 10);

  const grid = Array.from({ length: 101 }, () => Array(101).fill(0));

  for (let i = 1; i <= N; i++) {
    const [A, B, C, D] = lines[i].split(' ').map(Number);
    for (let x = A; x < B; x++) {
      for (let y = C; y < D; y++) {
        grid[x][y] += 1;
      }
    }
  }

  let area = 0;
  for (let x = 0; x <= 100; x++) {
    for (let y = 0; y <= 100; y++) {
      if (grid[x][y] > 0) {
        area += 1;
      }
    }
  }

  console.log(area);
}

main(require('fs').readFileSync('/dev/stdin', 'utf8'));

File: Atcoder_#318/ruby/b_overlapping_sheets.rb
def main
  n = gets.to_i
  grid = Array.new(101) { Array.new(101, 0) }

  n.times do
    a, b, c, d = gets.split.map(&:to_i)
    (a...b).each do |x|
      (c...d).each do |y|
        grid[x][y] += 1
      end
    end
  end

  area = 0
  (0..100).each do |x|
    (0..100).each do |y|
      area += 1 if grid[x][y] > 0
    end
  end

  puts area
end

main
